packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -660,18 +1115,229 @@ function recursivelyCommitLayoutEffects(
           }
         }
       }
-
-      if (enableScopeAPI) {
-        // TODO: This is a temporary solution that allowed us to transition away from React Flare on www.
-        if (flags & Ref && tag !== ScopeComponent) {
-          commitAttachRef(finishedWork);
-        }
-      } else {
-        if (flags & Ref) {
-          commitAttachRef(finishedWork);
-        }
+
+      if (enableScopeAPI) {
+        // TODO: This is a temporary solution that allowed us to transition away from React Flare on www.
+        if (flags & Ref && tag !== ScopeComponent) {
+          commitAttachRef(finishedWork);
+        }
+      } else {
+        if (flags & Ref) {
+          commitAttachRef(finishedWork);
+        }
+      }
+      break;
+    }
+  }
+}
+
+function iterativelyCommitLayoutEffects_begin(
+  subtreeRoot: Fiber,
+  finishedRoot: FiberRoot,
+) {
+  while (nextEffect !== null) {
+    const finishedWork: Fiber = nextEffect;
+    const firstChild = finishedWork.child;
+
+    if (
+      (finishedWork.subtreeFlags & LayoutMask) !== NoFlags &&
+      firstChild !== null
+    ) {
+      if (
+        enableProfilerTimer &&
+        enableProfilerCommitHooks &&
+        finishedWork.tag === Profiler
+      ) {
+        const prevProfilerOnStack = nearestProfilerOnStack;
+        nearestProfilerOnStack = finishedWork;
+
+        let child = firstChild;
+        while (child !== null) {
+          nextEffect = child;
+          iterativelyCommitLayoutEffects_begin(child, finishedRoot);
+          child = child.sibling;
+        }
+        nextEffect = finishedWork;
+
+        if ((finishedWork.flags & LayoutMask) !== NoFlags) {
```

Comment:
Shouldn't have any practical implications, but we check `LayoutMask` in the iterative branch and `Update | Callback` in the recursive branch. Might be worth updating the latter to also check `LayoutMask`.